### PR TITLE
Temporarily adding the container deployment

### LINKED ISSUE #97: Temporarily copy the language container deployment and activation from the Transformer Extension.
The container deployment and activation code will eventually be moved to the script-language-container-tool repo. 
Both extensions will be using this functionality for the container deployment.
In this ticket, we will just unify this code to make the migration easier. Also to be able to build the Extension Wrapper in the notebook-connector.

### PR DESCRIPTION
**Pull request by ahsimb**
closes #97

### CODE REVIEW

File: exasol_sagemaker_extension/utils/bucketfs_operations.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        bucketfs_name=bucketfs_name, connection_config=_bucketfs_connection)
+    _bucket_config = BucketConfig(
+        bucket_name=bucket, bucketfs_config=_bucketfs_config)
+    return BucketFSLocation(
+        bucket_config=_bucket_config,
+        base_path=PurePosixPath(path_in_bucket))
+
+
+def upload_model_files_to_bucketfs(
+        tmpdir_name: str, model_path: Path,
```
Comment (tkilias) [inline exasol_sagemaker_extension/utils/bucketfs_operations.py:44]: I think, that we don't need here, probably something we overlooked in the refactoring

File: exasol_sagemaker_extension/utils/bucketfs_operations.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    return file_path
+
+
+def create_tar_of_directory(path: Path, fileobj: BinaryIO):
+    with tarfile.open(name="model.tar.gz", mode="w|gz", fileobj=fileobj) as tar:
+        for subpath in path.glob("*"):
+            tar.add(name=subpath, arcname=subpath.name)
+
+
+def get_local_bucketfs_path(
```
Comment (tkilias) [inline exasol_sagemaker_extension/utils/bucketfs_operations.py:67]: I think that we don't need here, probably something we overlooked in the refactoring

File: exasol_sagemaker_extension/utils/bucketfs_operations.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        bucketfs_location: BucketFSLocation, model_path: str) -> PurePosixPath:
+    bucketfs_local_path = bucketfs_location.generate_bucket_udf_path(model_path)
+    return bucketfs_local_path
+
+
+def get_model_path(sub_dir: str, model_name: str) -> Path:
```
Comment (tkilias) [inline exasol_sagemaker_extension/utils/bucketfs_operations.py:73]: I think that we don't need here, probably something we overlooked in the refactoring

File: release_language_container.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+
```
Comment (tkilias) [inline release_language_container.sh:2]:
```suggestion
set -euo pipefail
```

File: release_language_container.sh [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+./build_language_container.sh
+
+mkdir .build_output/exported_container -p
```
Comment (tkilias) [inline release_language_container.sh:5]:
```suggestion
mkdir -p .build_output/exported_container
```

File: release_language_container.sh [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+cp .build_output/cache/exports/exasol_sagemaker_extension_container-release-*.tar.gz .build_output/exported_container/exasol_sagemaker_extension_container.tar.gz
```
Comment (tkilias) [inline release_language_container.sh:6]:
```suggestion
echo "Copying container to from .build_output/cache/exports to .build_output/exported_container"

cp .build_output/cache/exports/exasol_sagemaker_extension_container-release-*.tar.gz .build_output/exported_container/exasol_sagemaker_extension_container.tar.gz
```

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 7]
```diff
@@ -35,6 +35,9 @@ jobs:
       - name: Run Lua unit tests and static code analyzer
         run: poetry run poe lua-tests
 
+      - name: Release language container
+        run: ./release_language_container.sh
```
Comment (tkilias) [inline .github/workflows/release_droid_upload_github_release_assets.yml:39]: Thinking about, if we maybe should pass the export path (.build_output/exported_container/)  as a parameter to release_language_container.sh. Then we could define it as an environment variable in the GitHub Workflow and use it here in the upload.

File: doc/changes/changes_0.6.0.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+n/a
+  
+### Bug Fixes
+
+n/a
+  
+### Documentation
+
+- User manual is updated with instructions on how to use the language container deployment cli.
```
Comment (tkilias) [inline doc/changes/changes_0.6.0.md:23]:
```suggestion
```

That is not how we do changelog, the changelog contains the ticket.

File: doc/changes/changes_0.6.0.md [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+  
+### Refactoring
+
+n/a
```
Comment (tkilias) [inline doc/changes/changes_0.6.0.md:27]:
```suggestion
- #97: Improving the container deployment using the container deployer from [transformers-extension](https://github.com/exasol/transformers-extension/)  
```

File: doc/user_guide/user_guide.md [hunk 10]
```diff
@@ -40,32 +40,132 @@ within the scope of the project.
...
-where `ALIAS` is _PYTHON3_SME_, `LANGUAGE` is _python_ in the sagemaker-extension project.
-
-For more details please check [Adding New Packages to Existing Script Languages](https://docs.exasol.com/database_concepts/udf_scripts/adding_new_packages_script_languages.htm).
+This extension requires the installation of the language container for this 
+extension to run. It can be installed in two ways: Quick and Customized 
+installations
 
+#### Quick Installation
+The language container is downloaded and installed by executing the 
+deployment script below with the desired version. Make sure the version matches with your installed version of the
```
Comment (tkilias) [inline doc/user_guide/user_guide.md:51]: For the quick installation, we should actually use a proper default value for --version. However, this is a topic for a later ticket.

**Review by tkilias — CHANGES_REQUESTED** [review]

File: release_language_container.sh (hunk 5 above)
Comment (ahsimb) [reply release_language_container.sh:5]: Works either way, but OK

File: exasol_sagemaker_extension/utils/bucketfs_operations.py (hunk 1 above)
Comment (ahsimb) [reply exasol_sagemaker_extension/utils/bucketfs_operations.py:44]: Yes, that's right, I just copied the whole file.

File: .github/workflows/release_droid_upload_github_release_assets.yml (hunk 7 above)
Comment (ahsimb) [reply .github/workflows/release_droid_upload_github_release_assets.yml:39]: Aren't we going to make the release procedure similar to the Transformers Extension shortly? I so, why bother?

Comment (ahsimb) [reply .github/workflows/release_droid_upload_github_release_assets.yml:39]: OK, will do this for the learning purpose.

File: .github/workflows/release_droid_upload_github_release_assets.yml [hunk 11]
```diff
@@ -35,6 +38,9 @@ jobs:
...
```
Comment (tkilias) [inline .github/workflows/release_droid_upload_github_release_assets.yml:42]:
```suggestion
        run: ./release_language_container.sh "$RELEASE_DIR"
```

**Pull request merged**